astropy/utils/iers/iers.py

```diff
@@ -1025,15 +1026,20 @@ def expires(self):
     def _read_leap_seconds(cls, file, **kwargs):
         """Read a file, identifying expiration by matching 'File expires'"""
         expires = None
+        current = locale.setlocale(locale.LC_TIME)
         # Find expiration date.
         with get_readable_fileobj(file) as fh:
             lines = fh.readlines()
             for line in lines:
                 match = cls._re_expires.match(line)
                 if match:
-                    expires = Time.strptime(match.groups()[0], '%d %B %Y',
-                                            scale='tai', format='iso',
-                                            out_subfmt='date')
+                    try:
+                        locale.setlocale(locale.LC_TIME, 'C')
```

Comment:
Hmm, `locale` is quite explicit about not being thread safe. There is a suggested solution at https://stackoverflow.com/questions/18593661/how-do-i-strftime-a-date-object-in-a-different-locale, though I wonder if we are not better off just parsing it ourselves (as also suggested in various places). I think with a bit smarter regex this may just be possible........

Reply:
Yes indeed, the thing is that when using `_set_locale` here the tests are stuck in a deadlock...
Maybe another soution to parse the date would be better: 

> If, when coding a module for general use, you need a locale independent version of an operation that is affected by the locale (such as certain formats used with time.strftime()), you will have to find a way to do it without using the standard library routine.
https://docs.python.org/3/library/locale.html#background-details-hints-tips-and-caveats

Reply:
I think that just changing the `regex` to return the three elements and then doing a simply lookup on the first three characters of the month should do it:

https://github.com/astropy/astropy/blob/a68f4564d6c21ef90d82b22bfc6362cf67dd2d70/astropy/utils/iers/iers.py#L876